DelvEdit/src/com/interrupt/dungeoneer/editor/EditorApplication.java

```diff
@@ -433,18 +434,63 @@ public void loadMonsters() {
 		}
 	}
 
+	/** Creates an empty level with given `width` and `height`. */
 	public void createEmptyLevel(int width, int height) {
 		level = new Level(width, height);
+
+		Tile t = new Tile();
+		t.floorHeight = -0.5f;
+		t.ceilHeight = 0.5f;
+		level.setTile(width / 2, height / 2, t);
+
+		cleanEditorState();
+	}
+
+	/** Generates a level based on a `template` level. */
+	public void generateLevelFromTemplate(Level template) {
+		level.editorMarkers.clear();
+		level.entities.clear();
+		level.non_collidable_entities.clear();
+		level.static_entities.clear();
+
+		level.theme = template.theme;
+		level.generated = true;
+		level.dungeonLevel = 0;
+		level.crop(0, 0, 17 * 5, 17 * 5);
+		level.roomGeneratorChance = 0.4f;
+		level.roomGeneratorType = template.roomGeneratorType;
+		level.generate(Level.Source.EDITOR);
+
+		cleanEditorState();
+	}
+
+	/** Generates a single room based on a `template` level. */
+	public void generateRoomFromTemplate(Level template) {
+		level.editorMarkers.clear();
+		level.entities.clear();
+		level.non_collidable_entities.clear();
+		level.static_entities.clear();
+
+		Level generatedLevel = new Level(17, 17);
```

Comment:
This should probably be driven by `GenTheme.chunkTiles` and not hard coded.

Reply:
Yes. I thought grabbing that information would be *hacky*, but this way it's quite easy.